yt_dlp/YoutubeDL.py

```diff
@@ -3907,58 +3908,107 @@ def get_encoding(stream):
-        file_handler = urllib.request.FileHandler()
-
-        if not self.params.get('enable_file_urls'):
-            def file_open(*args, **kwargs):
-                raise urllib.error.URLError(
-                    'file:// URLs are explicitly disabled in yt-dlp for security reasons. '
-                    'Use --enable-file-urls to enable at your own risk.')
-            file_handler.file_open = file_open
-
-        opener = urllib.request.build_opener(
-            proxy_handler, https_handler, cookie_processor, ydlh, redirect_handler, data_handler, file_handler)
-
-        # Delete the default user-agent header, which would otherwise apply in
-        # cases where our custom HTTP handler doesn't come into play
-        # (See https://github.com/ytdl-org/youtube-dl/issues/1309 for details)
-        opener.addheaders = []
-        self._opener = opener
+
+        return proxies
+
+    @functools.cached_property
+    def cookiejar(self):
+        """Global cookiejar instance"""
+        return load_cookies(
+            self.params.get('cookiefile'), self.params.get('cookiesfrombrowser'), self)
+
+    @property
+    def _opener(self):
+        """
+        Deprecated: use YoutubeDL.urlopen() instead.
+        Get an urllib OpenerDirector from the Urllib handler.
+        """
+        handler = self._request_director.get_handler(rh_key='Urllib')
+        return handler._get_instance(cookiejar=self.cookiejar, proxies=self.proxies)
+
+    def urlopen(self, req):
+        """ Start an HTTP download """
+        if isinstance(req, str):
+            req = Request(req)
+        elif isinstance(req, urllib.request.Request):
+            # compat
+            req = urllib_req_to_req(req)
+        assert isinstance(req, Request)
+
+        # compat: Assume user:pass url params are basic auth
+        url, basic_auth_header = extract_basic_auth(req.url)
+        if basic_auth_header:
+            req.headers['Authorization'] = basic_auth_header
+        req.url = sanitize_url(url)
+
+        clean_proxies(proxies=req.proxies, headers=req.headers)
+        clean_headers(req.headers)
+
+        # TODO: do we want to use YoutubeDLError here or something else?
+        try:
+            return self._request_director.send(req)
+        except NoSupportingHandlers as e:
+            for ue in e.unsupported_errors:
+                if not (ue.handler and ue.msg):
+                    continue
+                if ue.handler.rh_key() == 'Urllib' and 'unsupported url scheme: "file"' in ue.msg.lower():
+                    raise YoutubeDLError(
+                        'file:// URLs are disabled by default in yt-dlp for security reasons. '
+                        'Use --enable-file-urls to enable at your own risk.')
+            raise
+        except SSLError as e:
+            if 'UNSAFE_LEGACY_RENEGOTIATION_DISABLED' in str(e):
+                raise YoutubeDLError('UNSAFE_LEGACY_RENEGOTIATION_DISABLED: Try using --legacy-server-connect') from e
```

Comment:
Pretty much just raising for the message that is more or less specific to YoutubeDL. 

YoutubeDLError would break through any `network_exceptions` catching, which may or may not be wanted.